R/ansi-hyperlink.R

```diff
@@ -426,3 +404,43 @@ ansi_hyperlink_types <- function() {
     )
   }
 }
+
+get_hyperlink_format <- function(type = c("run", "help", "vignette")) {
+  type <- match.arg(type)
+
+  key <- glue("hyperlink_{type}_url_format")
+  sprt <- ansi_hyperlink_types()[[type]]
+
+  custom_fmt <- get_config_chr(key)
+  if (is.null(custom_fmt)) {
+    if (identical(attr(sprt, "type"), "rstudio")) {
+      fmt_type <- "rstudio"
+    } else {
+      fmt_type <- "standard"
+    }
+  } else {
+    fmt_type <- "custom"
+  }
+
+  variable <- c(run = "code", help = "topic", vignette = "vignette")
+  fmt <- switch(
+    fmt_type,
+    custom = custom_fmt,
+    rstudio = glue("ide:{type}:{{{variable[type]}}}"),
+    standard = glue("x-r-{type}:{{{variable[type]}}}")
+  )
+  fmt
+}
+
+get_config_chr <- function(x, default = NULL) {
+  opt <- getOption(paste0("cli.", tolower(x)))
+  if (!is.null(opt)) {
+    stopifnot(is_string(opt))
+    return(opt)
+  }
+
+  env <- Sys.getenv(paste0("R_CLI_", toupper(x)), NA_character_)
+  if (!is.na(env)) return(env)
+
+  default
+}
```

Comment:
For everything I've touched, configuration is possible:

* Via an option or an env var. Context: in the Positron R console, options are easier for us (or at least that was true at some earlier point? but I have no reason to believe anything has changed). However, for R package dev tasks that run in a separate R process in an integrated terminal, env vars work much better.
* The option is consulted first. There was precedent for this.